packages/@lwc/engine-core/src/framework/stylesheet.ts

```diff
@@ -203,32 +195,39 @@ function getNearestNativeShadowComponent(vm: VM): VM | null {
     return owner;
 }
 
-export function createStylesheet(vm: VM, stylesheets: string[]): VNode | null {
+function createOrRemoveStylesheet(vm: VM, stylesheets: string[], insert: false): null;
+function createOrRemoveStylesheet(vm: VM, stylesheets: string[], insert: true): VNode | null;
+function createOrRemoveStylesheet(vm: VM, stylesheets: string[], insert: boolean) {
     const { renderMode, shadowMode } = vm;
     if (renderMode === RenderMode.Shadow && shadowMode === ShadowMode.Synthetic) {
         for (let i = 0; i < stylesheets.length; i++) {
-            insertGlobalStylesheet(stylesheets[i]);
+            toggleStyleSheet(stylesheets[i], insert);
         }
-    } else if (ssr || isHydrating()) {
+    } else if (ssr || vm.hydrated) {
```

Comment:
It turns out that what we care about is if the vm was hydrated in the first place, not whether it's currently hydrating.